### PR TITLE
Monorepo CI setup

### PR DESCRIPTION
**Pull request by hstove**
A number of changes:

- instead of symlinks / yarn link, i updated all the tsconfig.paths to just use the neighbor directory, so we don’t have to be building deps ever locally. 
- i made some changes to UI, mainly removing `skipNullChecks` and `suppressImplicitAnyIndexErrors`. This required a few small code changes
- Github actions for commitlint, keychain testing, "code checking" (lint, typecheck)
- Random config fixes to webpack and similar

### CODE REVIEW

File: packages/app/src/components/logo.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 
 import { Box } from '@blockstack/ui';
-import { BoxProps } from '@blockstack/ui/dist/box';
+import { BoxProps } from '@blockstack/ui';
```
Comment (aulneau) [inline packages/app/src/components/logo.tsx]: These two should be combined

File: packages/connect/src/react/components/link.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { Text, Box } from '@blockstack/ui';
-import { BoxProps } from '@blockstack/ui/dist/box/types';
+import { BoxProps } from '@blockstack/ui';
```
Comment (aulneau) [inline packages/connect/src/react/components/link.tsx]: These two should be combined

File: packages/ui/src/stack/index.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ export * from './types';
 
 const Stack = forwardRef<any, StackProps>(
   ({ isInline, children, align, justify, spacing = 2, shouldWrapChildren, ...rest }, ref) => {
-    const validChildren = Array.isArray(children) && children.filter(isValidElement);
+    const validChildren = Array.isArray(children) ? children.filter(isValidElement) : [];
```
Comment (aulneau) [inline packages/ui/src/stack/index.tsx:10]: Why was this changed?

File: packages/ui/src/use-clipboard/index.tsx [hunk 4]
```diff
@@ -18,14 +18,15 @@ const copyToClipboard = (value: string) => {
   el.style.left = '-9999px';
   document.body.appendChild(el);
 
-  const selected = document.getSelection().rangeCount > 0 ? document.getSelection().getRangeAt(0) : false;
+  const curSelection = document.getSelection();
```
Comment (aulneau) [inline packages/ui/src/use-clipboard/index.tsx:21]: We should maybe add in a typeof document !== 'undefined' here

**Review by aulneau — CHANGES_REQUESTED** [review]

File: packages/ui/src/stack/index.tsx (hunk 3 above)
Comment (hstove) [reply packages/ui/src/stack/index.tsx:10]: Because it caused a TS error after I disabled `skipNullChecks`. Technically, before, this would result in `validChildren` being either `false` or `[]`. That was causing errors down the file. With this change, `validChildren` can only be an array.

File: packages/ui/src/use-clipboard/index.tsx (hunk 4 above)
Comment (hstove) [reply packages/ui/src/use-clipboard/index.tsx:21]: Hm, not sure on that one - If we wanted those kind of checks, I'd rather change the `tsconfig` to not use `lib: dom`. But not worth it right now, IMO
